### PR TITLE
feat: platform status endpoint

### PR DESCRIPTION
**Pull request by shumkov**
## Issue being fixed or feature implemented

To monitor the network state we need to request a node status so we can see versions of various components, chain status, etc. 

## What was done?

- Introduced the platform `getStatus` DAPI endpoint
- Implemented partial status endpoint in Drive to share data with DAPI
- Added method to DAPI Client
- Updated documentation

## How Has This Been Tested?

Functional and unit tests

## Breaking Changes

None

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have added "!" to the title and described breaking changes in the corresponding section if my code contains any
- [x] I have made corresponding changes to the documentation if needed

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

File: packages/dapi-grpc/protos/platform/v0/platform.proto [hunk 1]
```diff
@@ -1001,3 +1002,92 @@ message GetPathElementsResponse {
...
+    }
+
+    message Time {
+      uint64 local = 1;
+      uint64 block = 2;
+      uint32 epoch = 3;
+    }
+
+    message Node {
+      string id = 1;
```
Comment (lklimek) [inline packages/dapi-grpc/protos/platform/v0/platform.proto:1049]: it's always named node_id, I would stick to it. I know it's inside node ;), but still

File: packages/dapi-grpc/protos/platform/v0/platform.proto [hunk 2]
```diff
@@ -1001,3 +1002,92 @@ message GetPathElementsResponse {
...
+      uint64 snapshot_chunks_count = 6;
+      uint64 backfilled_blocks = 7;
+      uint64 backfill_blocks_total = 8;
+    }
+
+    Version version = 1;
+    Node node = 2;
+    Chain chain = 3;
+    StateSync state_sync = 4;
+    Time time = 5;
```
Comment (lklimek) [inline packages/dapi-grpc/protos/platform/v0/platform.proto:1089]: what's that time?

File: packages/dapi-grpc/protos/platform/v0/platform.proto [hunk 3]
```diff
@@ -1001,3 +1002,92 @@ message GetPathElementsResponse {
...
+
+    message Time {
+      uint64 local = 1;
+      uint64 block = 2;
+      uint32 epoch = 3;
+    }
+
+    message Node {
+      string id = 1;
+      bytes pro_tx_hash = 2;
```
Comment (QuantumExplorer) [inline packages/dapi-grpc/protos/platform/v0/platform.proto:1050]: set as optional

File: packages/dapi-grpc/protos/platform/v0/platform.proto [hunk 4]
```diff
@@ -1001,3 +1002,92 @@ message GetPathElementsResponse {
...
+    message StateSync {
+      uint64 total_synced_time = 1;
+      uint64 remaining_time = 2;
+      uint32 total_snapshots = 3;
+      uint64 chunk_process_avg_time = 4;
+      uint64 snapshot_height = 5;
+      uint64 snapshot_chunks_count = 6;
+      uint64 backfilled_blocks = 7;
+      uint64 backfill_blocks_total = 8;
+    }
```
Comment (QuantumExplorer) [inline packages/dapi-grpc/protos/platform/v0/platform.proto:1083]: no need for state sync info

**Review by QuantumExplorer — APPROVED** [review]

**Pull request merged**